includes/validation/class-amp-validation-manager.php

```diff
@@ -519,6 +522,63 @@ function() {
 		self::$amp_admin_bar_item_added = true;
 	}
 
+	/**
+	 * Adds the debugging option links to the admin bar.
+	 *
+	 * @param WP_Admin_Bar $wp_admin_bar The admin bar to add the nodes to.
+	 * @param string       $amp_url The URL of the AMP endpoint.
+	 */
+	public static function add_debugging_option_nodes( &$wp_admin_bar, $amp_url ) {
+		$parent_node_id = 'amp-debug';
+		$wp_admin_bar->add_node(
+			[
+				'parent' => 'amp',
+				'id'     => $parent_node_id,
+				'title'  => esc_html__( 'Debugging options', 'amp' ),
+			]
+		);
+
+		$debugging_query_vars = [
+			AMP_Theme_Support::DISABLE_POST_PROCESSING_QUERY_VAR => esc_html__( 'Disable post-processing', 'amp' ),
+			AMP_Theme_Support::DISABLE_RESPONSE_CACHE_QUERY_VAR => esc_html__( 'Disable response cache', 'amp' ),
+			AMP_Theme_Support::PREVENT_REDIRECT_TO_NON_AMP_QUERY_VAR => esc_html__( 'Prevent redirect', 'amp' ),
+			AMP_Validation_Error_Taxonomy::REJECT_ALL_VALIDATION_ERRORS_QUERY_VAR => esc_html__( 'Reject all errors', 'amp' ),
+			AMP_Validation_Error_Taxonomy::ACCEPT_EXCESSIVE_CSS_ERROR_QUERY_VAR => esc_html__( 'Accept excessive CSS', 'amp' ),
+			'disable_amp' => esc_html__( 'Disable AMP', 'amp' ),
+			AMP_Style_Sanitizer::DISABLE_TREE_SHAKING_QUERY_VAR => esc_html__( 'Disable tree shaking', 'amp' ),
+		];
+
+		foreach ( $debugging_query_vars as $query_var => $title ) {
+			$wp_admin_bar->add_node(
+				[
+					'parent' => $parent_node_id,
+					'id'     => $query_var,
+					'title'  => esc_html( self::get_debugging_option_title( $title, $query_var ) ),
+					'href'   => add_query_arg( AMP_Theme_Support::AMP_FLAGS_QUERY_VAR, [ $query_var => '' ], $amp_url ),
+				]
+			);
+		}
+	}
+
+	/**
+	 * Gets the debugging option title.
+	 *
+	 * If the current URL has the debugging option query var present, the title starts with a 'check mark' emoji.
+	 * For example, if the URL has the 'Prevent redirect' query var present,
+	 * the title in the Admin Bar will be '✅ Prevent redirect'.
+	 *
+	 * @param string $title_text The title of the option, like 'Prevent redirect'.
+	 * @param string $query_var The query var of the option.
+	 * @return string The debugging option title, possible with a 'check mark' emoji at the beginning.
+	 */
+	public static function get_debugging_option_title( $title_text, $query_var ) {
+		if ( isset( $_GET[ AMP_Theme_Support::AMP_FLAGS_QUERY_VAR ][ $query_var ] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+			return '✅ ' . $title_text;
```

Comment:
Instead of this emoji, let's perhaps use these:

* “☑” (U+2611 Ballot Box with Check)
* “☐” (U+2610 Ballot Box)

When the option is enabled, the first should be used, and when the option is not enabled the second should be used. This will help with communicating that clicking on the link has the effect of toggling the option.

There will be a need to set the `font-family` explicitly to prevent the first one from being rendered as an emoji. So it can be wrapped in a `span`. They should end up looking like:

![image](https://user-images.githubusercontent.com/134745/67111000-e1580880-f188-11e9-9ee3-57bf54687c1c.png)

Reply:
Thanks, those look good!

Reply:
I'll change this to use those.

Reply:
Hi @westonruter, how does this look, with https://github.com/ampproject/amp-wp/pull/3448/commits/ef0720cdfaa78fa38623fe7ff39a9b43344290dc?
<img width="792" alt="new-emojis" src="https://user-images.githubusercontent.com/4063887/67115243-60f2d100-f1a3-11e9-9c30-9c2c25ea050d.png">

Reply:
Any reason why this can't be a checkbox input?